teuthology/paddles_queue.py

```diff
@@ -8,57 +6,32 @@
 from teuthology.config import config
 from teuthology import report
 
+
 log = logging.getLogger(__name__)
 
 
-def connect():
-    host = config.queue_host
-    port = config.queue_port
-    if host is None or port is None:
-        raise RuntimeError(
-            'Beanstalk queue information not found in {conf_path}'.format(
-                conf_path=config.teuthology_yaml))
-    return beanstalkc.Connection(host=host, port=port)
-
-
-def watch_tube(connection, tube_name):
-    """
-    Watch a given tube, potentially correcting to 'multi' if necessary. Returns
-    the tube_name that was actually used.
-    """
-    if ',' in tube_name:
-        log.debug("Correcting tube name to 'multi'")
-        tube_name = 'multi'
-    connection.watch(tube_name)
-    connection.ignore('default')
-    return tube_name
-
-
-def walk_jobs(connection, tube_name, processor, pattern=None):
-    """
-    def callback(jobs_dict)
-    """
-    log.info("Checking Beanstalk Queue...")
-    job_count = connection.stats_tube(tube_name)['current-jobs-ready']
-    if job_count == 0:
-        log.info('No jobs in Beanstalk Queue')
-        return
+def stats_queue(machine_type):
+    stats = report.get_queue_stats(machine_type)
+    stats = report.get_queue_stats(machine_type)
+    if stats['paused'] is None:
+        log.info("%s queue is currently running with %s jobs queued",
+                 stats['name'],
+                 stats['count'])
+    else:
+        log.info("%s queue is paused with %s jobs queued",
+                 stats['name'],
+                 stats['count'])
 
-    # Try to figure out a sane timeout based on how many jobs are in the queue
-    timeout = job_count / 2000.0 * 60
-    for i in range(1, job_count + 1):
-        print_progress(i, job_count, "Loading")
-        job = connection.reserve(timeout=timeout)
-        if job is None or job.body is None:
-            continue
-        job_config = yaml.safe_load(job.body)
-        job_name = job_config['name']
-        job_id = job.stats()['id']
-        if pattern is not None and pattern not in job_name:
-            continue
-        processor.add_job(job_id, job_config, job)
-    end_progress()
-    processor.complete()
+
+def update_priority(machine_type, priority, user):
```

Comment:
usually we'd want to adjust the priority of a particular run, rather than all of a user's jobs. automatic scheduled jobs, for example, are all queued as the same user, and it's helpful to differentiate their priority.

I'd suggest taking user xor run name for this command

Reply:
Sure!